Dockerfile

```diff
@@ -0,0 +1,24 @@
+ARG PG_MAJOR=15
+FROM postgres:$PG_MAJOR
+ARG PG_MAJOR
+
+COPY . /tmp/pg_embedding
+
+# Locale settings
+RUN apt-get clean && apt-get update && \
+    apt-get install -y locales && \
+    localedef -i en_US -c -f UTF-8 -A /usr/share/locale/locale.alias en_US.UTF-8
+
+ENV LANG en_US.utf8
+
+RUN apt-get install -y --no-install-recommends build-essential postgresql-server-dev-$PG_MAJOR && \
+        cd /tmp/pg_embedding && \
+        make clean && \
+        make OPTFLAGS="" && \
+        make install && \
+        mkdir /usr/share/doc/pg_embedding && \
+        cp LICENSE README.md /usr/share/doc/pg_embedding && \
+        rm -r /tmp/pg_embedding && \
+        apt-get remove -y build-essential postgresql-server-dev-$PG_MAJOR && \
+        apt-get autoremove -y && \
+        rm -rf /var/lib/apt/lists/*
```

Comment:
Instead of cleaning up things afterwards, I would probably split this into a [multi-stage build](https://docs.docker.com/build/building/multi-stage/#use-multi-stage-builds):
- build pg_embedding
- copy the extension from the previous stage to a fresh`postgres:$PG_MAJOR` container